### PR TITLE
[1.19.3] Add block atlas config to register forge:white texture

### PR DESCRIPTION
**Pull request by XFactHD**
This PR adds an atlas config for the block atlas that registers the `forge:white` texture to be stitched onto the block atlas.
This texture is regularly used as a placeholder in block and item models or in cases where only a flat color is needed but the involved parts of the rendering system require a texture.

### CODE REVIEW

**Comment by LexManos** [comment]
All data should be genned.

File: src/main/java/net/minecraftforge/common/data/SpriteSourceProvider.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package net.minecraftforge.common.data;
+
+import com.mojang.serialization.*;
```
Comment (Minecraftschurli) [inline src/main/java/net/minecraftforge/common/data/SpriteSourceProvider.java]: Wildcard import

**Review by Minecraftschurli — CHANGES_REQUESTED** [review]

**Review by Minecraftschurli — APPROVED** [review]

File: src/main/java/net/minecraftforge/common/data/SpriteSourceProvider.java (hunk 1 above)
Comment (XFactHD) [reply src/main/java/net/minecraftforge/common/data/SpriteSourceProvider.java]: Of course I had to miss some...

**Comment by TheCurle** [comment]
Don't forget your licenses.

File: src/main/java/net/minecraftforge/common/data/SpriteSourceProvider.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    protected final void gather(BiConsumer<ResourceLocation, List<SpriteSource>> consumer)
+    {
+        addSources();
+        atlases.forEach((atlas, srcList) -> consumer.accept(atlas, srcList.sources));
+    }
+
+    protected abstract void addSources();
+
+
+
```
Comment (sciwhiz12) [inline src/main/java/net/minecraftforge/common/data/SpriteSourceProvider.java]: Please remove this extra whitespace. This should only be a single blank line.

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]

File: src/main/java/net/minecraftforge/common/data/SpriteSourceProvider.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.PackType;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+public abstract class SpriteSourceProvider extends JsonCodecProvider<List<SpriteSource>>
```
Comment (KnightMiner) [inline src/main/java/net/minecraftforge/common/data/SpriteSourceProvider.java:28]: A javadoc might be useful here, to explain what a sprite source does, or at least link relevant classes

**Review by sciwhiz12 — APPROVED** [review]

**Comment by marchermans** [comment]
Has this been renewed on top of the compatibility test fix?

**Comment by XFactHD** [comment]
This PR is blocked by https://github.com/MinecraftForge/MinecraftForge/pull/9222 as the addition of the atlas config currently breaks all textures stitched onto the block atlas when run in a Forgedev environment.

**Pull request merged**